src/enums.ts

```diff
@@ -0,0 +1,15 @@
+export enum BarRectType {
```

Comment:
use `const enum` instead of `enum` everywhere if you really don't need a regular `enum` features.